File: app/src/main/java/com/spidroid/starry/repositories/CommentRepository.java
package com.spidroid.starry.repositories;

import com.google.android.gms.tasks.Task;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.FieldValue;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.Query;
import com.google.firebase.firestore.QuerySnapshot;

import java.util.HashMap;
import java.util.Map;

public class CommentRepository {
    private final FirebaseFirestore db = FirebaseFirestore.getInstance();

    public Task<QuerySnapshot> getCommentsForPost(String postId) {
        return db.collection("posts")
                .document(postId)
                .collection("comments")
                .orderBy("timestamp", Query.Direction.ASCENDING)
                .get();
    }
// ... (داخل كلاس CommentRepository)

    public Task<DocumentReference> addComment(String postId, Map<String, Object> commentData) {
        return db.collection("posts")
                .document(postId)
                .collection("comments")
                .add(commentData);
    }

    public Task<Void> toggleCommentLike(String postId, String commentId, String userId, boolean isLiked) {
        DocumentReference commentRef = db.collection("posts").document(postId).collection("comments").document(commentId);
        Map<String, Object> updates = new HashMap<>();
        updates.put("likeCount", FieldValue.increment(isLiked ? 1 : -1));
        updates.put("likes." + userId, isLiked ? true : FieldValue.delete());
        return commentRef.update(updates);
    }

    public Task<Void> deleteComment(String postId, String commentId) {
        return db.collection("posts")
                .document(postId)
                .collection("comments")
                .document(commentId)
                .delete();
    }

    public Task<Void> updatePostReplyCount(String postId, int increment) {
        return db.collection("posts")
                .document(postId)
                .update("replyCount", FieldValue.increment(increment));
    }
    // يمكنك إضافة دوال أخرى هنا لاحقًا (مثل إضافة تعليق، حذفه، إلخ)
}

File: app/src/main/java/com/spidroid/starry/repositories/CommentRepository.kt
package com.spidroid.starry.repositories

import com.google.android.gms.tasks.Task
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.firestore.Query
import com.google.firebase.firestore.QuerySnapshot
import kotlinx.coroutines.tasks.await
import javax.inject.Inject
import javax.inject.Singleton

// --- ١. تمت إضافة هذه التعليقات التوضيحية ---
@Singleton
class CommentRepository @Inject constructor(
    private val firestore: FirebaseFirestore
) {

    private val postsCollection = firestore.collection("posts")

    fun getCommentsForPost(postId: String): Task<QuerySnapshot> {
        return postsCollection.document(postId).collection("comments")
            .orderBy("timestamp", Query.Direction.DESCENDING)
            .get()
    }

    fun addComment(postId: String, commentData: Map<String, Any>): Task<Void> {
        val commentRef = postsCollection.document(postId).collection("comments").document()
        return commentRef.set(commentData)
    }

    fun toggleCommentLike(postId: String, commentId: String, userId: String, isLiked: Boolean): Task<Void> {
        val commentRef = postsCollection.document(postId).collection("comments").document(commentId)
        val updateData = mapOf("likes.$userId" to isLiked)
        return commentRef.update(updateData)
    }

    fun deleteComment(postId: String, commentId: String): Task<Void> {
        return postsCollection.document(postId).collection("comments").document(commentId).delete()
    }

    fun updatePostReplyCount(postId: String, increment: Long): Task<Void> {
        val postRef = postsCollection.document(postId)
        return firestore.runTransaction { transaction ->
            val snapshot = transaction.get(postRef)
            val newReplyCount = (snapshot.getLong("replyCount") ?: 0) + increment
            transaction.update(postRef, "replyCount", newReplyCount)
            null
        }
    }
}